### PR TITLE
Feature/correlating bpmn messages

### PR DESCRIPTION
**Pull request by luis-soares-ritense**

### CODE REVIEW

File: using-valtimo/process/sendingmessages.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Sending messages in BPMN
+
+Valtimo provides a way to send messages to another process/sub-process allowing not only
+correlation but association between processes and a Valtimo case.<br/>
```
Comment (marijnritense) [inline using-valtimo/process/sendingmessages.md:4]: Why are we using `<br/>`? There's no need most of the time.

Also not sure what this sentence is meant to convey. Maybe the structure as a whole to:

When modeling a process, correlating a message (e.g. to start another process) does not do anything by default. Valtimo offers several methods to facilitate message correlation, which is intended to cover several different events and possible variations depending on the goal. These can be separated into starting a process (Message Start Events) and message catch events (Message Boundary Events and Message Intermediate Catch Events).

<<list of different use cases and the methods that they correspond to here>>

### How to use

<<explanation on how to use these methods in a BPMN>>

File: using-valtimo/process/correlating-messages.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Correlating messages in BPMN
+
+When modeling a process, correlating a message (e.g. to start another process) does not do anything by default. Valtimo
+offers several methods to facilitate message correlation. These can be separated into starting a process
+(Message Start Events) and message catch events (Message Boundary Events and Message Intermediate Catch Events).
```
Comment (ivo-ritense) [inline using-valtimo/process/correlating-messages.md:5]:
```suggestion
(Message Start Events) and receiving messages in a running process (Message Boundary Events and Message Intermediate Catch Events).
```

File: using-valtimo/process/correlating-messages.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+## How to use
+
+Valtimo provides several methods that can be used inside a BPMN, accessible through the `correlationService` bean. Which
+method to use depends on the use case. For more information on the separate use cases see the two sections below.
+
+These methods can be used in expressions applied to message throw events like this:
+
+![intermediate-throw-example](img/intermediate-throw-event.png)
+
+The first argument always corresponds to the message the is supposed to send. In this example, there should also be a
```
Comment (ivo-ritense) [inline using-valtimo/process/correlating-messages.md:16]:
```suggestion
The first argument is the key of the message that should be send. In this example, there should also be a
```

Comment (marijnritense) [reply using-valtimo/process/correlating-messages.md:16]:
```suggestion
The first argument is the key of the message that should be sent. In this example, there should also be a
```

**Comment by marijnritense** [comment]
Still missing: release notes, and this PR should be merged to the correct branch

**Review by ivo-ritense — APPROVED** [review]

**Pull request merged**